core/server/master/src/main/java/alluxio/master/file/meta/MountTable.java

```diff
@@ -321,49 +349,76 @@ public Map<String, MountInfo> getMountTable() {
   }
 
   /**
-   * @param uri the Alluxio uri to check
+   * build the MountTableTrie based on given InodeTree. This will trigger the recovering of
+   * MountTableTrie.
+   *
+   * @param inodeTree the rootInode set in MountTableTrie
+   */
+  public void buildMountTableTrie(InodeTree inodeTree) throws Exception {
+    try (LockResource r = new LockResource(mWriteLock)) {
+      mMountTableTrie.recoverFromInodeTreeAndMountPoints(inodeTree,
+          mState.getMountTable().keySet());
+    }
+  }
+
+  /**
+   * Enable the MountTableTrie based on the given rootInode. It will not trigger the recovering
+   * of MountTableTrie.
+   * @param rootInode the rootInode set in MountTableTrie
+   */
+  public void buildMountTableTrie(InodeView rootInode) {
```

Comment:
```
Precondition.checkNotNull(rootInode, "Root inode cannot be null");
```
Also add a precondition check that the trie doesn't already have a root inode set.